pkg/hostagent/event_poster.go

```diff
@@ -0,0 +1,169 @@
+		eventSubmitTimeMap: make(map[string]time.Time),
+	}
+}
+
+// Init Event Recorder for poster object
+func (agent *HostAgent) initEventRecorder(kubeClient *kubernetes.Clientset) record.EventRecorder {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(
+		&typedcorev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
+	component := "aci-containers-host"
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: component})
+	return recorder
+}
+
+// Submit an event using kube API with message attached
+func (agent *HostAgent) submitEvent(pod *v1.Pod, message string, dropReason string) error {
+	agent.log.Info("Posting event ", message)
+	ref, err := reference.GetReference(scheme.Scheme, pod)
+	if err != nil {
+		agent.log.Error("Returning ", err)
+		return err
+	}
+	if dropReason == "" {
+		dropReason = "Dropped"
+	}
+	if agent.poster != nil && agent.poster.recorder != nil {
+		agent.poster.recorder.Event(ref, v1.EventTypeWarning, dropReason, message)
+	}
+	return nil
+}
+
+// Check if given PacketEvent should be ignored
+func (agent *HostAgent) shouldIgnore(packetEvent PacketEvent) bool {
+	// ignore if the given packetDrop is out of date
+	logTime, err := time.Parse(time.UnixDate, packetEvent.TimeStamp)
+	currTime := time.Now()
+	if (err != nil) || (currTime.Sub(logTime).Minutes() > float64(agent.config.DropLogExpiryTime)) {
+		agent.log.Debug("Ignoring expired event")
+		return true
+	}
+	// ignore if the event has been posted within the defined time period
+	key := packetEvent.SourceIP + packetEvent.DestinationIP
+	lastPostedTime := agent.poster.eventSubmitTimeMap[key]
+	repeatEventIntervalMinutes := float64(agent.config.DropLogRepeatIntervalTime)
+	if !lastPostedTime.IsZero() && (logTime.Sub(lastPostedTime).Minutes() <= repeatEventIntervalMinutes) {
+		agent.log.Debug("Ignoring event as it occurred within repeatInterval")
+		return true
+	}
+	return false
+}
+
+// Construct packet drop message
+func getPacketDropMessage(srcIp string, dstIp string) string {
+	var buffer bytes.Buffer
+	buffer.WriteString("Packet from ")
+	buffer.WriteString(srcIp)
+	buffer.WriteString(" to ")
+	buffer.WriteString(dstIp)
+	buffer.WriteString(" was dropped")
+	return buffer.String()
+}
+
+// Handle the given PacketDrop, return error if api server does not work
+func (agent *HostAgent) processPacketEvent(packetEvent PacketEvent) error {
+	if packetEvent.SourceIP == "" || packetEvent.DestinationIP == "" {
+		return nil
+	}
+	if agent.shouldIgnore(packetEvent) {
+		return nil
+	}
+	agent.indexMutex.Lock()
+	srcPodKey, srcOk := agent.podIpToName[packetEvent.SourceIP]
+	agent.indexMutex.Unlock()
+	agent.log.Debug("srcPodKey for ", packetEvent.SourceIP, ": ", srcPodKey)
+	if !srcOk {
+		agent.log.Debug(packetEvent.SourceIP, "may not be a pod or not a local pod")
+	} else {
+		obj1, srcExists, err := agent.podInformer.GetStore().GetByKey(srcPodKey)
+		if err == nil {
+			if srcExists {
```

Comment:
can you check obj also here along with srcExists

Reply:
sure